external/optiga_config.h

```diff
@@ -0,0 +1,125 @@
+    /** @brief OPTIGA CRYPT RSA sign feature enable/disable macro */
+    #define OPTIGA_CRYPT_RSA_SIGN_ENABLED
+    /** @brief OPTIGA CRYPT RSA verify sign feature enable/disable macro */
+    #define OPTIGA_CRYPT_RSA_VERIFY_ENABLED
+    /** @brief OPTIGA CRYPT RSA Encrypt feature enable/disable macro */
+    #define OPTIGA_CRYPT_RSA_ENCRYPT_ENABLED
+    /** @brief OPTIGA CRYPT RSA Decrypt feature enable/disable macro */
+    #define OPTIGA_CRYPT_RSA_DECRYPT_ENABLED
+    /** @brief OPTIGA CRYPT RSA pre-master feature enable/disable macro */
+    #define OPTIGA_CRYPT_RSA_PRE_MASTER_SECRET_ENABLED
+    /** @brief OPTIGA CRYPT RSA SSA with SHA512 as digest feature enable/disable macro */      
+    #define OPTIGA_CRYPT_RSA_SSA_SHA512_ENABLED      
+    /** @brief OPTIGA CRYPT symmetric encrypt feature enable/disable macro */
+    #define OPTIGA_CRYPT_SYM_ENCRYPT_ENABLED
+    /** @brief OPTIGA CRYPT symmetric decrypt feature enable/disable macro */
+    #define OPTIGA_CRYPT_SYM_DECRYPT_ENABLED
+    /** @brief OPTIGA CRYPT HMAC feature enable/disable macro */
+    #define OPTIGA_CRYPT_HMAC_ENABLED
+    /** @brief OPTIGA CRYPT HKDF feature enable/disable macro */
+    #define OPTIGA_CRYPT_HKDF_ENABLED
+    /** @brief OPTIGA CRYPT symmetric generate key feature enable/disable macro */
+    #define OPTIGA_CRYPT_SYM_GENERATE_KEY_ENABLED
+    /** @brief OPTIGA CRYPT generate auth code feature enable/disable macro */
+    #define OPTIGA_CRYPT_GENERATE_AUTH_CODE_ENABLED
+    /** @brief OPTIGA CRYPT HMAC verify feature enable/disable macro */
+    #define OPTIGA_CRYPT_HMAC_VERIFY_ENABLED
+    /** @brief OPTIGA CRYPT clear AUTO state feature enable/disable macro */
+    #define OPTIGA_CRYPT_CLEAR_AUTO_STATE_ENABLED   
+
+    /** @brief OPTIGA COMMS shielded connection feature.
+     *         To disable the feature, undefine the macro
+     */
+    #define OPTIGA_COMMS_SHIELDED_CONNECTION
+
+    /** @brief Default reset protection level for OPTIGA CRYPT and UTIL APIs */
+    #define OPTIGA_COMMS_DEFAULT_PROTECTION_LEVEL           OPTIGA_COMMS_FULL_PROTECTION
+    //#define OPTIGA_COMMS_DEFAULT_PROTECTION_LEVEL           OPTIGA_COMMS_NO_PROTECTION
+
+  /** @brief Default reset type in optiga_comms_open.             \n
+     *         Cold Reset - (0) : This is applicable if the host platform has GPIO option for RST and VDD.    \n
+     *         Soft Reset - (1) : This is applicable if the host platform doesn't have GPIO options for VDD and RST.  \n
+     *         Warm Reset - (2) : This is applicable if the host platform doesn't have GPIO option for VDD. \n
+     *         Any other value will lead to error
+     */
+    #define OPTIGA_COMMS_DEFAULT_RESET_TYPE     (1U)
+    
+    /** @brief NULL parameter check.
+     *         To disable the check, undefine the macro
+     */
+    #define OPTIGA_LIB_DEBUG_NULL_CHECK
+    /** @brief Maximum number of instance registration */
+    #define OPTIGA_CMD_MAX_REGISTRATIONS                (0x06)
```

Comment:
Could this be 1, does that have any impact on size?

Reply:
Saves 288 bytes :smile:

Reply:
@NickeZ actually 0x01 fails, 0x02 works. With 0x01, the second one fails:

```
    util = optiga_util_create(OPTIGA_INSTANCE_ID_0, optiga_lib_callback, NULL);
    if (NULL == util) {
        util_log("couldn't create optiga util");
        return false;
    }

    crypt = optiga_crypt_create(OPTIGA_INSTANCE_ID_0, optiga_lib_callback, NULL);
    if (NULL == crypt) {
        util_log("couldn't create optiga crypt");
        return false;
    }

```